### PR TITLE
Fix Conda CI

### PR DESCRIPTION
**Pull request by vorj**
Currently CI jobs using conda are failed due to conflict packages.
This PR fixes this.

- use newer `numpy` to build `faiss-cpu`
- install `pytorch` when testing `faiss-cpu`
    - to find correct `pytorch` package, `pytorch` channel is set at `conda build`

### CODE REVIEW

**Comment by vorj** [comment]
The job for Windows will be fixed by #1876 .

File: conda/faiss/meta.yaml [hunk 1]
```diff
@@ -65,7 +65,7 @@ outputs:
         - make  # [not win]
       host:
         - python {{ python }}
-        - numpy 1.11.*
+        - numpy >=1.11.*
```
Comment (beauby) [inline conda/faiss/meta.yaml:68]: We need to build against numpy `1.11`, as this is the earliest version of numpy we support, and we use the C API. If we relax the constraint here, packages generated will only be compatible with numpy versions newer than whichever conda-build ends up selecting at build time.

File: .circleci/config.yml [hunk 2]
```diff
@@ -93,9 +93,8 @@ jobs:
           name: Conda build
           command: |
             conda install -y -q conda-build
-            conda install -c pytorch pytorch
             cd conda
-            conda build faiss --python 3.7
+            conda build faiss --python 3.7 -c pytorch
```
Comment (beauby) [inline .circleci/config.yml:97]: These changes are still desirable. Would you mind rebasing your PR?

File: conda/faiss/meta.yaml (hunk 1 above)
Comment (vorj) [reply conda/faiss/meta.yaml]: I see. I confirmed that you fixed this part on #1894, so I've reverted this change.

File: conda/faiss/meta.yaml [hunk 3]
```diff
@@ -40,7 +40,7 @@ outputs:
       host:
         - mkl =2018
       run:
-        - mkl >=2018
+        - mkl >=2018,<2021
```
Comment (vorj) [inline conda/faiss/meta.yaml:43]: `mkl-2021.2.0-haa95532_296` doesn't contain `mkl_rt.dll` (it contains `mkl_rt.1.dll` instead). We need to constrain `mkl` as `<2021` if `faiss` should be built with `mkl=2018` (linked to `mkl_rt.dll`) .

**Pull request closed without merging**